Use dockerCmd when possible

hqhq opened the pull request:
Addresses: #14603

integration-cli/docker_cli_daemon_experimental_test.go (hqhq)
integration-cli/docker_cli_daemon_test.go (hqhq)
integration-cli/docker_cli_diff_test.go (hqhq)
integration-cli/docker_cli_events_test.go (hqhq)
integration-cli/docker_cli_events_unix_test.go (hqhq)
integration-cli/docker_cli_exec_test.go (hqhq)
integration-cli/docker_cli_exec_unix_test.go (hqhq)
integration-cli/docker_cli_experimental_test.go (hqhq)
integration-cli/docker_cli_export_import_test.go (hqhq)
integration-cli/docker_cli_help_test.go (hqhq)
integration-cli/docker_cli_history_test.go (hqhq)
integration-cli/docker_cli_images_test.go (hqhq)
integration-cli/docker_cli_import_test.go (hqhq)
integration-cli/docker_cli_info_test.go (hqhq)
integration-cli/docker_cli_inspect_test.go (hqhq)
integration-cli/docker_cli_kill_test.go (hqhq)

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

hqhq commented:
ping @calavera @cpuguy83 Since you reviewed other related PRs.

vdemeester commented on integration-cli/docker_cli_exec_test.go:
```diff
@@ -240,44 +176,20 @@ func (s *DockerSuite) TestExecTtyCloseStdin(c *check.C) {
 	}
 
-	defer func() {
-		cmd := exec.Command(dockerBinary, "kill", id)
```
@hqhq what about this one ?

hqhq replied:
I think this defer function is for container cleanup which is not needed now, so I removed it. Maybe I shouldn't change any logics in this PR?

hqhq replied:
Yeah, I think I shouldn't change logic here, updated.

vdemeester replied:
Yeah I think so too :wink:

hqhq commented:
I force updated my branch for almost an hour, but codes in this PR is still not updated, and Jenkins is not retriggered, don't know what's going on...

vdemeester commented:
@hqhq yeah something is weird, the commit hash on the PR has not been update :sweat:

calavera commented:
probably some glitch in the GitHub matrix. Try adding/removing something trivial and pushing again.

runcom commented on integration-cli/docker_cli_events_test.go:
```diff
@@ -134,10 +128,9 @@ func (s *DockerSuite) TestEventsLimit(c *check.C) {
-	if exitCode != 0 || err != nil {
-		c.Fatalf("Failed to get events with exit code %d: %s", exitCode, err)
+	out, exitCode := dockerCmd(c, "events", "--since=0", fmt.Sprintf("--until=%d", daemonTime(c).Unix()))
+	if exitCode != 0 {
```
nit, this execCode is always != 0 here since there's no error here (otherwise dockerCmd should have failed before)

runcom commented on integration-cli/docker_cli_events_test.go:
```diff
@@ -171,11 +164,10 @@ func (s *DockerSuite) TestEventsContainerEventsSinceUnixEpoch(c *check.C) {
-	out, exitCode, err := runCommandWithOutput(eventsCmd)
-	if exitCode != 0 || err != nil {
-		c.Fatalf("Failed to get events with exit code %d: %s", exitCode, err)
+	if exitCode != 0 {
```
as above

runcom commented on integration-cli/docker_cli_events_test.go:
```diff
@@ -217,10 +209,9 @@ func (s *DockerSuite) TestEventsImageUntagDelete(c *check.C) {
-	if exitCode != 0 || err != nil {
-		c.Fatalf("Failed to get events with exit code %d: %s", exitCode, err)
+	out, exitCode := dockerCmd(c, "events", "--since=0", fmt.Sprintf("--until=%d", daemonTime(c).Unix()))
+	if exitCode != 0 {
```
ditto, exitCode always != 0

runcom commented on integration-cli/docker_cli_history_test.go:
```diff
@@ -47,7 +46,7 @@ RUN echo "Z"`,
 	}
 
-	out, exitCode, err := runCommandWithOutput(exec.Command(dockerBinary, "history", "testbuildhistory"))
+	out, exitCode, err := dockerCmdWithError(c, "history", "testbuildhistory")
```
I think `dockerCmd` should suffice here (w/o checking exitCode and err outside)

runcom commented on integration-cli/docker_cli_history_test.go:
```diff
@@ -67,16 +66,14 @@ RUN echo "Z"`,
 func (s *DockerSuite) TestHistoryExistentImage(c *check.C) {
-	historyCmd := exec.Command(dockerBinary, "history", "busybox")
-	_, exitCode, err := runCommandWithOutput(historyCmd)
+	_, exitCode, err := dockerCmdWithError(c, "history", "busybox")
```
just dockerCmd

runcom commented on integration-cli/docker_cli_history_test.go:
```diff
@@ -67,16 +66,14 @@ RUN echo "Z"`,
 func (s *DockerSuite) TestHistoryNonExistentImage(c *check.C) {
-	historyCmd := exec.Command(dockerBinary, "history", "testHistoryNonExistentImage")
-	_, exitCode, err := runCommandWithOutput(historyCmd)
+	_, exitCode, err := dockerCmdWithError(c, "history", "testHistoryNonExistentImage")
```
`dockerCmd`

hqhq replied:
We are expecting command to be failed, so I think here need a `dockerCmdWithError`.

vdemeester replied:
yep, we are checking for non-nil so yes, ``dockerCmdWithError`` on this one.

runcom replied:
perfect ok, can we remove the check on exitCode == 0 then since if err == nil exitCode should be always 0?

runcom commented on integration-cli/docker_cli_info_test.go:
```diff
@@ -10,8 +9,7 @@ import (
 func (s *DockerSuite) TestInfoEnsureSucceeds(c *check.C) {
-	versionCmd := exec.Command(dockerBinary, "info")
-	out, exitCode, err := runCommandWithOutput(versionCmd)
+	out, exitCode, err := dockerCmdWithError(c, "info")
```
just `dockerCmd` here

runcom commented on integration-cli/docker_cli_inspect_test.go:
```diff
@@ -43,15 +41,9 @@ func (s *DockerSuite) TestInspectDefault(c *check.C) {
+	dockerCmd(c, "run", "--name=busybox", "-d", "busybox", "true")
 
-	_, exitCode, err := runCommandWithOutput(inspectCmd)
+	out, exitCode, err := dockerCmdWithError(c, "inspect", "busybox")
```
`dockerCmd` :)

hqhq committed:
dockerCmd when possible

Addresses: #14603

integration-cli/docker_cli_daemon_experimental_test.go (hqhq)
integration-cli/docker_cli_daemon_test.go (hqhq)
integration-cli/docker_cli_diff_test.go (hqhq)
integration-cli/docker_cli_events_test.go (hqhq)
integration-cli/docker_cli_events_unix_test.go (hqhq)
integration-cli/docker_cli_exec_test.go (hqhq)
integration-cli/docker_cli_exec_unix_test.go (hqhq)
integration-cli/docker_cli_experimental_test.go (hqhq)
integration-cli/docker_cli_export_import_test.go (hqhq)
integration-cli/docker_cli_help_test.go (hqhq)
integration-cli/docker_cli_history_test.go (hqhq)
integration-cli/docker_cli_images_test.go (hqhq)
integration-cli/docker_cli_import_test.go (hqhq)
integration-cli/docker_cli_info_test.go (hqhq)
integration-cli/docker_cli_inspect_test.go (hqhq)
integration-cli/docker_cli_kill_test.go (hqhq)

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq commented:
@runcom Thanks for reviewing, except one comment, I addressed all your comments, PTAL, thanks.

runcom commented:
One minor comment left, otherwise LGTM

hqhq commented:
@runcom Done, thanks.

calavera commented:
So clean, thank you!

LGTM
